### ISSUE TITLE
CLI option to prefer lowest version of dependencies (for test buids)

### ISSUE THREAD

**Issue by thaven** [dlang/dub#1762]
To test the correctness of version requirements, it would be useful to have a CLI option to force selection of the lowest matching version of any dependency, much like [Composer](https://getcomposer.org/doc/03-cli.md#update-u)'s ˋ--prefer-lowestˋ.

Such option used in automatic builds may catch mistakes like introducing a call to a function only present in package-x versions `˜>1.2.0`, while still specifying `˜>1.0.0` as a dependency.

If using this option, DUB still selects a later version than the lowest matching version according to the package's own dependencies (i.e. a later version is indirectly required), a warning should be produced.

**Comment by joseph-wakeling-frequenz** [comment]
Agreed, this is an important feature.  I would also suggest a `dub downgrade` option to select the lowest versions manually.
